src/lib.rs

```diff
@@ -0,0 +1,94 @@
+pub mod resource_id {
+    use std::fs;
+    use std::path::Path;
+    use std::io::{BufReader, BufRead};
+    use crc32fast::Hasher;
+    use log::trace;
+    pub fn compute_id<P: AsRef<Path>>(
+        file_size: usize,
+        file_path: P) -> i64 {
```

Comment:
If we need `i64` only for passing into Kotlin, I would create some adapter on that side.
We should use `u32` here if possible.